compiler/next/test/resolution/testCanPass.cpp

```diff
@@ -349,18 +366,208 @@ static void test6() {
                                    StringType::get(context),
                                    p);
 
-  auto stringQT = QualifiedType(QualifiedType::VALUE,
+  auto stringQT = QualifiedType(QualifiedType::VAR,
                                 StringType::get(context));
-  auto cStringQT = QualifiedType(QualifiedType::VALUE,
+  auto cStringQT = QualifiedType(QualifiedType::VAR,
                                  CStringType::get(context));
-  auto bytesQT = QualifiedType(QualifiedType::VALUE,
+  auto bytesQT = QualifiedType(QualifiedType::VAR,
                                BytesType::get(context));
   CanPassResult r;
   r = canPass(paramString, stringQT); assert(passesAsIs(r));
   r = canPass(paramString, cStringQT); assert(passesParamNarrowing(r));
   r = canPass(paramString, bytesQT); assert(doesNotPass(r));
 }
 
+static void test7() {
+  printf("test7\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  // test that we can pass a child class to its parent class type
+  // but that we can't do the inverse.
+
+  ID emptyId;
+  std::vector<CompositeType::FieldDetail> emptyFields;
+
+  auto parentName = UniqueString::build(context, "Parent");
+  auto childName = UniqueString::build(context, "Child");
+  auto basicObj = BasicClassType::getObjectType(context);
+  auto basicParent = BasicClassType::get(context, emptyId, parentName,
+                                         basicObj, emptyFields);
+  auto basicChild = BasicClassType::get(context, emptyId, childName,
+                                        basicParent, emptyFields);
+
+  auto borrowed = ClassTypeDecorator(ClassTypeDecorator::BORROWED_NONNIL);
```

Comment:
Do these also need to be using a `ClassTypeDecorator::get`?

Reply:
ClassTypeDecorator just wraps an enum up nicely so we can have methods on it. No reason to have it be a query or a get method IMO. We could have a get method as a style matter though.